Blockzilla/Modules/WebView/WebViewController.swift

```diff
@@ -82,6 +82,27 @@ class WebViewController: UIViewController, WebController {
     func goBack() { browserView.goBack() }
     func goForward() { browserView.goForward() }
     func reload() { browserView.reload() }
+    
+    @available(iOS 9, *)
+    func requestDesktop() {
+        
+        guard let currentItem = browserView.backForwardList.currentItem else {
+            return
+        }
+    
+        browserView.customUserAgent = UserAgent.getDesktopUserAgent()
+        
+        if currentItem.url != currentItem.initialURL {
+            // Reload the initial URL to avoid UA specific redirection
+            browserView.load(URLRequest(url: currentItem.initialURL, cachePolicy: .reloadIgnoringLocalAndRemoteCacheData, timeoutInterval: 60))
+        } else {
+            reload() // Reload the current URL. We cannot use loadRequest in this case because it seems to leverage caching.
+        }
+        
+        // Unset the desktopUserAgent
+        browserView.customUserAgent = UserAgent.browserUserAgent
```

Comment:
Will this send a user back to the mobile page if they click a link?

Reply:
Yep, this is the default behavior in Safari, so I mirrored that. Do you think we should handle it differently?

Reply:
Nope! I wasn't aware :) Thanks for the clarification!